blockchain/blocks/src/tipset.rs

```diff
@@ -62,6 +62,19 @@ pub struct Tipset {
     key: TipsetKeys,
 }
 
+impl From<FullTipset> for Tipset {
+    fn from(full_tipset: FullTipset) -> Self {
+        Tipset::new(
+            full_tipset
+                .blocks
+                .into_iter()
+                .map(|block| block.header)
+                .collect(),
+        )
+        .unwrap()
```

Comment:
Why are you unwrapping here instead of returning a Result?

Reply:
Yeah this looks like it should implement a Tryfrom instead

Reply:
the point of this is that the same validation is done in FullTipset as Tipset, so you can just construct a Tipset without having to go through the Tipset constructor (which does all the validation again)